src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -556,10 +583,9 @@ int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
         /* detach the matched node from tree */
         gavl_tree_delete_internal(tree_ptr, dnode);
 
-        /* free object using user-provided free function */
-        if (tree_ptr->gavl_free_fn)
-            tree_ptr->gavl_free_fn((void *) dnode->val);
-        MPL_free(dnode);
+        /* add removed node into remove list */
+        dnode->next = tree_ptr->remove_list;
+        tree_ptr->remove_list = dnode;
 
```

Comment:
Can we move the following lines into `gavl_tree_delete_internal`?
```
        /* add removed node into remove list */
        dnode->next = tree_ptr->remove_list;
        tree_ptr->remove_list = dnode;
```

Moreover, it might be more suitable to replace `gavl_tree_delete_internal` with `gavl_tree_remove_node_internal`.